src/SAWScript/Crucible/JVM/Builtins.hs

```diff
@@ -463,6 +487,8 @@ setupPrePointsTos mspec cc env pts mem0 = foldM doPointsTo mem0 pts
                  Just x -> pure x
              rhs' <- traverse (injectSetupVal . MS.SetupTerm) tts
              doEntireArrayStore sym mem lhs' rhs'
+        JVMPointsToArray _loc _lhs Nothing ->
+          pure mem -- This should probably never occur in the prestate section.
```

Comment:
Is it worth warning if this happens?

Reply:
A `jvm_modifies` declaration in the pre-state section is basically meaningless. I could make that an error. But probably what I should do is make all the `jvm_modifies` declarations be recorded in the post-state section, no matter where they are written. I think that is closest to the behavior that a reasonable user would expect.

If I make that change, then I can make this particular line of code into a panic.

Reply:
I decided to just make `jvm_modifies` in the pre-state be an error. I extended `test_jvm_setup_errors` in the test suite to also test this new error condition, and that line of code is now a panic.